sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/StorageScope.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.Storage;
+using Microsoft.Azure.Management.Storage.Models;
+using Microsoft.Rest;
+
+namespace Azure.Messaging.ServiceBus.Tests
+{
+    /// <summary>
+    ///  Provides a dynamically created Azure blob container instance which exists only in the context
+    ///  of the scope; disposal removes the instance.
+    /// </summary>
+    ///
+    /// <seealso cref="System.IAsyncDisposable" />
+    ///
+    public sealed class StorageScope : IAsyncDisposable
```

Comment:
I don't think we should add this. Creating a storage account is a solved problem. We do it in many places using the test-resources. I don't see why SB needs a 4 files worth of custom logic to do this.

Reply:
See discussion with @jsquire - the concern was that we are not being consistent with ServiceBusScope, which is in turn using the same pattern as EventHubScope. I tend to agree that this feature is not really worth the weight of its complexity, but we already have it written and working, so don't really see too much harm in being different here.

If we want to standardize, I would suggest we do this for both SB and EH and limit the Scope classes to simply creating queues/topics/event hubs.

Reply:
>so don't really see too much harm in being different here from other T2 libraries.

There are multiple benefits outside of the initial cost of writing:
1. Support costs. Now you need to support the new logic (2 copies) and keep fixing things that "just work" in other places.
2. Knowledge transfer. A new person coming to the project would know our test infrastructure but won't know this thing.
3. Refactoring/improvement opportunities. Keeping things consistent allows them to be improved more easily and in a centralized manner without having to dig into every library's custom logic.

>If we want to standardize, I would suggest we do this for both SB and EH and limit the Scope classes to simply creating queues/topics/event hubs.

I'm ok with this.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/23418

I don't think we need to block this PR on fixing this issue, as it is an already existing issue across both libraries (I am just making it even more of an issue here 😆)

Reply:
> don't think we need to block this PR on fixing this issue,

I don't see why not. The fix is trivial.

Reply:
Fixed